hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/KafkaOffsetGen.java

```diff
@@ -476,7 +476,7 @@ private Map<TopicPartition, Long> getGroupOffsets(KafkaConsumer consumer, Set<To
         fromOffsets.put(topicPartition, committedOffsetAndMetadata.offset());
       } else {
         LOG.warn("There are no commits associated with this consumer group, starting to consume from latest offset");
-        fromOffsets = consumer.endOffsets(topicPartitions);
+        fromOffsets = consumer.beginningOffsets(topicPartitions);
```

Comment:
Could you add a comment saying this will only happen once per consumer grp ? Otherwise the offsets would be committed

strange it was reading from beginning by default..

Reply:
Added comments. It is very odd and confusing.